CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -76,4 +77,67 @@ private static function validateDates(array $params) {
     }
   }
 
+
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array $params
+   *  Request Parameters
+   *
+   * @throws \Exception
+   */
+  private static function validateReviewFields(array &$params) {
+    $reviewFieldsLegalOptions = [
+      'id' => FILTER_VALIDATE_INT,
+      'weight' => FILTER_VALIDATE_INT,
+      'required' => FILTER_VALIDATE_BOOLEAN
+    ];
+    $reviewFields = CRM_Utils_Array::value('review_fields', $params);
+    if (empty($reviewFields)) {
+      return;
+    }
+
+    $filteredReviewFields = [];
+    foreach ($reviewFields as $reviewFieldOptions) {
+      $invalidReviewFieldOptions = [];
+      $validReviewFieldOptions = array_filter(
+        $reviewFieldOptions,
+        function($value, $key) use ($reviewFieldsLegalOptions, &$invalidReviewFieldOptions) {
+          if (!isset($reviewFieldsLegalOptions[$key])) {
+            return FALSE;
+          }
+
+          if (filter_var($value, $reviewFieldsLegalOptions[$key], FILTER_NULL_ON_FAILURE) !== NULL) {
+            return TRUE;
+          }
+          $invalidReviewFieldOptions[] = $key;
+        },
+        ARRAY_FILTER_USE_BOTH
+      );
+      if (count($validReviewFieldOptions) === count($reviewFieldsLegalOptions)) {
+        continue;
+      }
+
+      $missingReviewFieldOptions = array_diff(
+        array_keys($reviewFieldsLegalOptions),
+        array_merge($invalidReviewFieldOptions, array_keys($validReviewFieldOptions))
+      );
+      $exceptionPhrases = [
+        !empty($invalidReviewFieldOptions) ? "invalid values passed for [%1]" : "",
+        !empty($missingReviewFieldOptions) ? "required field" . (
+          count($missingReviewFieldOptions) > 1 ? "s [%2] are" : " [%2] is"
+        ) . " missing" : ""
+      ];
+
+      throw new Exception(ts(implode(' and ', array_filter($exceptionPhrases)), [
+        1 => implode(', ', $invalidReviewFieldOptions),
+        2 => implode(', ', $missingReviewFieldOptions),
+      ]));
+
+      $filteredReviewFields[] = $validReviewFieldOptions;
```

Comment:
I don't think this is necessary. Also the code will never get to this stage because once the review fields are valid, the code will never get to this place and the loop continues. 
```php
     if (count($validReviewFieldOptions) === count($reviewFieldsLegalOptions)) {
        continue;
      }
```
Then, exceptions will definitely be thrown, so code will not get to this stage.
So if the review fields are valid, technically there is a bug here because `filteredReviewFields` will be empty and the line below `$params['review_fields'] = $filteredReviewFields;` will set the review fields to empty.

Reply:
You are right. I have removed this bit